project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -665,6 +585,23 @@ public CompletableFuture<List<Result<MeasurementId, ErrorCode>>> updateAll(
       List<MeasurementMetadata> measurementMetadataList, ProjectId projectId) {
     List<Result<MeasurementId, ErrorCode>> results;
 
+    if (!measurementMetadataList.isEmpty() && measurementMetadataList.get(
+        0) instanceof ProteomicsMeasurementMetadata) {
+      List<ProteomicsMeasurementMetadata> metadata = new ArrayList<>();
+      for (MeasurementMetadata measurementMetadata : measurementMetadataList) {
+        metadata.add((ProteomicsMeasurementMetadata) measurementMetadata);
+      }
+      try {
+        results = updateAllPxp(metadata, projectId);
+      } catch (MeasurementRegistrationException e) {
+        log.error("Measurement update failed.", e);
+        return CompletableFuture.completedFuture(List.of(Result.fromError(e.reason)));
+      }
+      return CompletableFuture.completedFuture(results);
```

Comment:
Why do we need to work with completable future here? o.O

Reply:
Async service API

Reply:
Yes but why do we need to work asynchronously here? Can the validation not be done in one sequential way? This way we could fail earlier as we do not have to wait for all the validations to go through. Did I overlook something?

As this was this way before your PR and the PR is already large and addressing another problem I am fine with leaving it for now.

Reply:
The API is of async, non-blocking, nature to enable the UI to continue and not block the user.